tests/test_qmath.py

```diff
@@ -0,0 +1,45 @@
+# MIT License
+#
+# Copyright (c) 2022 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import perceval as pcvl
+from perceval.utils.qmath import exponentiation_by_squaring
+
+
+def test_exponentiation():
+    assert exponentiation_by_squaring(12, 1), 12
+    assert exponentiation_by_squaring(8, 2), 64
+    assert exponentiation_by_squaring(52, 13), 20325604337285010030592
+    bs = pcvl.BasicState("|1,0,1,0,0>")
+    assert bs**3, bs*bs*bs
+    sv = pcvl.StateVector(pcvl.BasicState("|1,0,1,0,0>")) + pcvl.StateVector(pcvl.BasicState("|1,0,0,1,0>"))
+    assert sv**2, sv*sv
+    assert sv**5, sv*sv*sv*sv*sv
+    sv = pcvl.StateVector(pcvl.BasicState('|1,0,4,0,0,2,0,1>')) - pcvl.StateVector(pcvl.BasicState('|1,0,3,0,2,1,0,1>'))
+    assert sv**2, sv*sv
+    assert sv**7, sv*sv*sv*sv*sv*sv*sv*sv*sv
```

Comment:
I think you mean `assert a == b` and not `assert a, b` which only checks if the tuple `(a, b)` is `True` which it always is.
I checked-out your branch, replaced comma with == and your test isn't passing.

Also, could you check it works for states with annotated photons, and on probability distributions (`BSDistribution`, `SVDistribution`) ?
Thanks!